src/test/components/elements/panels/SelectedEventDetails/countGenders.test.tsx

```diff
@@ -0,0 +1,20 @@
+import { EventBookingDTO } from "../../../../../IsaacApiTypes";
+import { countGenders } from "../../../../../app/components/elements/panels/SelectedEventDetails";
+import { mockCancelledEventBooking, mockEventBookings } from "../../../../../mocks/data";
+
+describe("countGenders", () => {
+  it("returns an object with all genders set to 0 when passed an empty array", () => {
+    const result = countGenders([]);
+    expect(result).toEqual({ male: 0, female: 0, other: 0, preferNotToSay: 0, unknown: 0 });
+  });
+
+  it("should return the correct count of genders when passed an array of Event Bookings", () => {
+    const result = countGenders(mockEventBookings);
+    expect(result).toEqual({ male: 1, female: 1, other: 1, preferNotToSay: 1, unknown: 1 });
+  });
+
+  it("should not count bookings with invalid bookingStatus", () => {
+    const results = countGenders([...mockEventBookings, mockCancelledEventBooking] as EventBookingDTO[]);
```

Comment:
Instead of doing this casting here, the right thing to do is to set the type of `mockCancelledEventBooking` to `EventBookingDTO`. You can then go and remove the casting to `EventBookingDTO` in every `mockCancelledEventBooking` usage.

This is a pattern that I keep seeing all over the codebase and it's wrong, because we're misusing castings when we should simply be setting the variables to the right types.

Reply:
This is part of the `Date` type issue previously discussed. Some of it is fixed in another branch, and we have a ticket brought into this test to address it also.

Reply:
Mmmmm... I don't understand how this is part of the `Date` type issue. You can simply set the type of `mockCancelledEventBooking` to `EventBookingDTO` and then remove the casting to `EventBookingDTO` in the places where `mockCancelledEventBooking` is being used (less than a handful) and everything should work straight away.